src/charm.py

```diff
@@ -590,6 +340,49 @@ def _ingress_config(self) -> dict:
             },
         }
 
+    def get_receiver_url(self, protocol: ReceiverProtocol):
```

Comment:
didn't we have a very similar piece of code in the tracing library (`get_endpoint`)? Can you check if we can avoid duplicating that by lifting it out to a utility method we can invoke from here?

Reply:
The one we have in the lib `get_endpoint` gets the receivers endpoints from the `TracingRequirerAppData` databag. This data gets published using the `get_receiver_url` in charm code. So, this function is actually the source.

Reply:
uh right, I may be confusing it from a point in time when this logic was living in the charm lib. We refactored it out of it at some point I guess. Thanks for checking :)